pkg/tagset/hash_generator.go

```diff
@@ -135,3 +135,110 @@ func (g *HashGenerator) Hash(tb *HashingTagsAccumulator) uint64 {
 
 	return hash
 }
+
+// Dedup2 removes duplicates from two tags accumulators. Duplicate tags are removed, so at the end
+// tag each tag is present once in either l or r, but not both at the same time.
+//
+// First, duplicates are removed from l. Then duplicates are removed from r, including any tags that
+// are already present in l. Can move tags from one accumulator to another.
```

Comment:
No, the direction does not matter. There are only two instances of HTA used with this function. We copy tags to them, compute the hashes and deduplicate, and then copy out if the tags are new and we need to store them. Accumulators are then cleared and re-used for another metric. Which accumulator ends up holding which tags affects only our ability to share them effectively between contexts (so we want to keep lower cardinality tags from origin detection separate from potentially higher cardinality tags sent by the client), but otherwise we don't need to make any distinction between the two.